File: svws-asd/src/main/java/de/svws_nrw/asd/validate/klassen/ValidatorKow00KlassenOrganisationsformWeiterbildung.java
package de.svws_nrw.asd.validate.klassen;

import java.util.function.Supplier;

import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import de.svws_nrw.transpiler.annotations.AllowNull;
import jakarta.validation.constraints.NotNull;

/**
 * Validator KOW00: Prüft, ob für die Klasse eine Organisationsform Weiterbildend angegeben wurde.
 */
public final class ValidatorKow00KlassenOrganisationsformWeiterbildung extends Validator {

	private final @NotNull Supplier<@AllowNull Long> _idWeiterbildendOrganisationsform;

	/**
	 * @param idWeiterbildendOrganisationsform	ID
	 * @param kontext							Kontext
	 */
	public ValidatorKow00KlassenOrganisationsformWeiterbildung(
			final @NotNull Supplier<@AllowNull Long> idWeiterbildendOrganisationsform,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		this._idWeiterbildendOrganisationsform = idWeiterbildendOrganisationsform;

		_validatoren.add(new ValidatorKow01KlassenOrganisationsformWeiterbildung(getNotNullSupplierLong(idWeiterbildendOrganisationsform), kontext));
	}

	@Override
	protected boolean pruefe() {

		if (null == _idWeiterbildendOrganisationsform.get()) {
			addFehler(0, "Organisationsform der Klasse: Kein Wert vorhanden.");
			return false;
		}

		return true;
	}

}

File: svws-webclient/core/src/asd/validate/klassen/ValidatorKow00KlassenOrganisationsformWeiterbildung.ts
import type { Supplier } from '../../../java/util/function/Supplier';
import { Class } from '../../../java/lang/Class';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';
import { ValidatorKow01KlassenOrganisationsformWeiterbildung } from '../../../asd/validate/klassen/ValidatorKow01KlassenOrganisationsformWeiterbildung';

export class ValidatorKow00KlassenOrganisationsformWeiterbildung extends Validator {

	private readonly _idWeiterbildendOrganisationsform: Supplier<number | null>;


	/**
	 * @param idWeiterbildendOrganisationsform	ID
	 * @param kontext							Kontext
	 */
	public constructor(idWeiterbildendOrganisationsform: Supplier<number | null>, kontext: ValidatorKontext) {
		super(kontext);
		this._idWeiterbildendOrganisationsform = idWeiterbildendOrganisationsform;
		this._validatoren.add(new ValidatorKow01KlassenOrganisationsformWeiterbildung(this.getNotNullSupplierLong(idWeiterbildendOrganisationsform), kontext));
	}

	protected pruefe(): boolean {
		if (null === this._idWeiterbildendOrganisationsform.get()) {
			this.addFehler(0, "Organisationsform der Klasse: Kein Wert vorhanden.");
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.klassen.ValidatorKow00KlassenOrganisationsformWeiterbildung';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.klassen.ValidatorKow00KlassenOrganisationsformWeiterbildung', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorKow00KlassenOrganisationsformWeiterbildung>('de.svws_nrw.asd.validate.klassen.ValidatorKow00KlassenOrganisationsformWeiterbildung');

}

export function cast_de_svws_nrw_asd_validate_klassen_ValidatorKow00KlassenOrganisationsformWeiterbildung(obj: unknown): ValidatorKow00KlassenOrganisationsformWeiterbildung {
	return obj as ValidatorKow00KlassenOrganisationsformWeiterbildung;
}
